JPA Map Storage with CRDB: ConcurrentTransactionsTest failures

### Describe the bug

**org.keycloak.testsuite.model.ConcurrentTransactionsTest** presents failures/errors with `CockroachDB`:

```
[ERROR]   ConcurrentTransactionsTest.persistClient » RunOnServer java.lang.NullPointerException
```

- Required JPA stores to reproduce the failures: **realms**

The specific point where this happens follows:

 ```
 Caused by: java.lang.NullPointerException
	at org.keycloak.models.map.user.MapUserProvider.getUserByUsername(MapUserProvider.java:490)
	at org.keycloak.testsuite.model.ConcurrentTransactionsTest.tearDownRealm(ConcurrentTransactionsTest.java:284)
	at org.keycloak.testsuite.model.ConcurrentTransactionsTest.persistClient(ConcurrentTransactionsTest.java:192)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.keycloak.testsuite.rest.TestingResourceProvider.runModelTestOnServer(TestingResourceProvider.java:853)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
```

In this specific case, the `realm` that is being supplied to `getUserByUsername` is `null` - the test calls `session.realms().getRealmByName("original")` and this is returning `null`, even though the entry **exists** in the database (which can be verified by setting a breakpoint in `MapRealmProvider.getRealmByName()` and querying the DB right before hibernate does so.

### Version

main

### Expected behavior

Tests should run without failures or errors



### Actual behavior

_No response_

### How to Reproduce?

1- Setup CockroachDB according to the instructions in https://github.com/keycloak/keycloak/discussions/9596
2- Run the test with the correct parameters, enabling JPA storage for **realms** only:

```
cd testsuite/integration-arquillian

mvn clean install -Pmap-storage -Dpageload.timeout=90000 -Dkeycloak.realm.map.storage.provider=jpa \
-Dkeycloak.map.storage.connectionsJpa.url=jdbc:postgresql://localhost:26257/keycloak \
-Dkeycloak.map.storage.connectionsJpa.user=keycloak -Dkeycloak.map.storage.connectionsJpa.password=pass \
-Dtest=ConcurrentTransactionsTest
```

### Anything else?

_No response_

--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/ConcurrentTransactionsTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/ConcurrentTransactionsTest.java
@@ -189,7 +189,7 @@ public class ConcurrentTransactionsTest extends AbstractTestRealmKeycloakTest {
 
             });
         } finally {
-            tearDownRealm(session, "user1", "user2");
+            KeycloakModelUtils.runJobInTransaction(session.getKeycloakSessionFactory(), kcSession -> tearDownRealm(kcSession, "user1", "user2"));
         }
     }
 

